### ISSUE TITLE
Reset code is always 123456

### ISSUE THREAD

**Issue by SCOTPAUL** [projectgoav/studeso#49]
When I go to reset my password, I always get the code 123456. Why is this happening?

**Comment by mmckay99** [comment]
It seems to be hardcoded in:

`u = UserReset.objects.get_or_create(username=username, code=123456)[0]` (regviews.py)

Do you think if we change that to a random number then it'll work?

**Comment by SCOTPAUL** [comment]
I have no idea. @projectgoav did you not write any random code generation?

**Comment by mmckay99** [comment]
I changed it to a random sequence of numeric digits of length 5 - 10, and it seems to work.

**Comment by SCOTPAUL** [comment]
If I'm logged out and click "Forgot my password" it tells me a reset email has been sent out too when it hasn't. You need to manually click "Send another" or whatever to get to the reset form.

**Comment by mmckay99** [comment]
Yeah I noticed this too- but I'm finding it really hard to debug the registration views...

**Comment by SCOTPAUL** [comment]
They are pretty confusing since I've never really looked at them before. I'm beginning to think registration-redux might've been a good idea...

**Comment by mmckay99** [comment]
We need @projectgoav !

**Comment by SCOTPAUL** [comment]
He's away just now I think. I suppose we have until Friday to fix it, as long as we avoid demoing the password reset we might be alright.

**Comment by mmckay99** [comment]
Oh yeah that's true! I can't imagine why he'd want us to demo the password reset anyway...

**Comment by ghost** [comment]
Yeah if you really want that to work on it then that's fine but I'm happy to demo what we have just now, plus as well as webapps we have that final JOOSE lab and me and Paul have maths stuff due the end of the week

**Comment by projectgoav** [comment]
Password reset works, i've just hardcoded it to make sure that email sending and stuff works. I'd like to make it a 6 digit code: 100000 -> 999999 since we're not handling a huge volume of users and I'm sure I check the length but off hand, I can't remember.

As for resending emails and stuff, it's a quick 2min fix. I'll pop that in along with random codes when I get home.

**Comment by SCOTPAUL** [comment]
OK. As long as we have code generation and that Integrity Error fixed I think we're ready to demo.

**Comment by mmckay99** [comment]
Integrity Error is being tricky!

Update: should be fixed (https://github.com/projectgoav/studeso/commit/96a05b00f0941c1ade5b111ccb3959842698470c)

**Comment by SCOTPAUL** [comment]
My solution would be adding this in `clean()`

``` python
username = self.cleaned_data.get('username')
        if Tag.objects.filter(name=username).exists():
            self.add_error('username', "A tag already exists on this site with that name. Please choose another.")
```

**Comment by SCOTPAUL** [comment]
Ah, didn't realise you'd already fixed it.

**Comment by mmckay99** [comment]
Strangely the add_error didn't work- it still proceeded to create the user (and collide with a tag name)... I had to throw a validation error, but it looks nice enough I think.

**Comment by projectgoav** [comment]
Registration stuff fixed in: 731ecc36c5c0120d8d28d6e555c2a09a2e306dac
